cranelift/codegen/src/loop_analysis.rs

```diff
@@ -232,15 +236,11 @@ impl LoopAnalysis {
         for lp in self.loops().rev() {
             stack.extend(
                 cfg.pred_iter(self.loops[lp].header)
-                    .filter(
-                        |BlockPredecessor {
-                             inst: pred_inst, ..
-                         }| {
-                            // We follow the back edges
-                            domtree.dominates(self.loops[lp].header, *pred_inst, layout)
-                        },
-                    )
-                    .map(|BlockPredecessor { block, .. }| block),
+                    .filter(|pred| {
+                        // We follow the back edges
+                        domtree.dominates(self.loops[lp].header, pred.inst, layout)
+                    })
+                    .map(|pred| pred.block),
```

Comment:
I think this would be a little clearer if we had a comment above the `stack.extend`, something like "Push all predecessors of this header that this header dominates onto the stack."